2226/2/versus.rb

```diff
@@ -0,0 +1,192 @@
+require 'terminal-table'
+require 'fuzzy_match'
+require 'optparse'
+require 'russian'
+require 'yaml'
+
+# This class is used for storing data
+class DataStorage
+  def self.show_all_data
+    Dir.glob('rap-battles/*').each_with_object({}) do |file_name, hash|
+      hash[file_name] = File.read(file_name)
+    end
+  end
+
+  def self.list_all_rapers
+    filter_rapers.flatten.uniq
+  end
+
+  def self.filter_rapers_from_the_list(file_name)
+    filter = file_name.sub(/ротив_/, '').sub(/_\(.*/, '')
+    filter.sub(%r/^rap-battles\/_/, '').sub(/_$/, '')
+  end
+
+  def self.filter_rapers
+    show_all_data.inject([]) do |array, (file_name, _file_content)|
+      filtered_data = filter_rapers_from_the_list(file_name).split('_п')
+      array << filtered_data
+    end
+  end
+end
+
+# This class is used for single raper methods
+class Raper
+  SWEAR_WORDS = YAML.load_file('config.yml')['SWEAR_WORDS_ARRAY'].join('|')
+
+  def self.avg_number_of_battles(name)
+    DataStorage.show_all_data.inject(0) do |counter, (file_name, _file_content)|
```

Comment:
sum

Reply:
Can't use sum and count because of if operator

Reply:
Can't use sum and count because of if operator